PhysicsTools/JetMCAlgos/plugins/GenHFHadronMatcher.cc

```diff
@@ -358,7 +362,7 @@ std::vector<int> GenHFHadronMatcher::findHadronJets(const reco::GenParticleColle
 
       // Scanning the chain starting from the hadron
       int hadronIndex = analyzeMothers(
-          thisParticle, topDaughterQId, topBarDaughterQId, hadMothersCand, hadMothersIndices, nullptr, -1);
+          thisParticle, topDaughterQId, topBarDaughterQId, hadMothersCand, hadMothersIndices, analyzedParticles, -1);
```

Comment:
This also changes the behavior and the suggestion above would preserve the present behavior.